app/controllers/hiring_staff/base_controller.rb

```diff
@@ -20,32 +20,32 @@ def check_session
   end
 
   def check_terms_and_conditions
-    redirect_to terms_and_conditions_path unless current_user&.accepted_terms_and_conditions?
+    redirect_to terms_and_conditions_path unless current_publisher&.accepted_terms_and_conditions?
   end
 
-  def current_user
+  def current_publisher
     return if current_session_id.blank?
 
-    @current_user ||= User.find_or_create_by(oid: current_session_id)
+    @current_publisher ||= Publisher.find_or_create_by(oid: current_session_id)
```

Comment:
If I am logged in as a jobseeker using the session id, then this line will create a Publisher. How can we ensure that ids of Publishers and Jobseekers do not collide?

Reply:
> this sounds ok but does it not mean that really should rename hiringstaff which is used in file/folder naming a lot to publisher too? 'We now have multiple distinct types of users, so the original naming of
> hiring staff members as "user" is confusing.' - is the naming of hiring staff members as publishers not also confusing? if you get my drift. maybe im overthinking :|

Excellent point @alexbowen – I was thinking of renaming that too, but possibly as a separate follow-on PR so that this one doesn't get too overwhelming! 👍